### ISSUE TITLE
BeaconReinforcement

### ISSUE THREAD

**Issue by DavidxRussell** [dawson-vilamaa/BeaconWaypoints#19]
I think this idea could be the future of SMP and I would definitely pay for this. My idea is for this  project to expand to BeaconReinforcement, a protection add on.
See what reinforcement is here: https://github.com/trevor99z/Reinforcement 

The reinforcement region size would be determined by the Beacon pyramid. The amount of “breaks” would be determined by what blocks were used in the pyramid’s construction, all configurable in config. Permitted users can build/break/access with the region with ease, non-permitted would have to break through the reinforcement.

Along with the current GUI, there would be another tab (maybe a chest plate icon) where a user could add trusted members, purchase BeaconAlarms (with items, exp, or Vault. similar with how teleport costs work now), and also access block history. I think block history would be dope for dynamics as users could see what enemies broke at what time so they’re prepared to fight back.

If someone destroys a Beacon the region would start to “decay” by the reinforcement dropping one block-break every server tick. The destruction of the Beacon would also trigger the BeaconAlarms. During BeaconDecay all users within the region would receive the red tint that the world border gives you when you’re outside.
This would create a bury dynamic, where users would bury their beacons under layers of glass for protection. Griefers would target Beacons to make their raid easier.
Other details about BeaconDecay..
Would be cool if there was an animation. Maybe thunder?

BeaconAlarms include Weakness, Slowness, Blindness, Nausea, Spectral Arrow Outline, and Mining Fatigue.

TLDR;
BeaconReinforcement
- Reinforcement sister plugin
- Members
- Block History
- Decay
- Discord Hooks
- BeaconAlarms

**Comment by dawson-vilamaa** [comment]
While these are good ideas, they are outside of the scope of this plugin which focuses on fast travel using beacons.
